lib/webmachine/adapters/rack.rb

```diff
@@ -95,6 +92,14 @@ def call(env)
         rack_res.finish
       end
 
+      private
+      def build_webmachine_request(rack_req, headers)
+        Webmachine::Request.new(rack_req.request_method,
+                                rack_req.url,
+                                headers,
+                                RequestBody.new(rack_req))
+      end
+
```

Comment:
I'm always a bit hesitant about overriding private methods, especially when you're overriding actual behaviour, not just "nil" or empty. I can see why it's tempting here though, as it's such a small change in behaviour.
The only thing that differs between the two classes is the setting of the routing_tokens. Given I've suggested making routing_tokens immutable, you'd need to duplicate the call to the constructor in the sub class, as you won't be able to change the routing_tokens after creation. What if you created a method `routing_tokens (rack_req)` that by default returned nil, and was overriden in RackMapped to return the custom one? That way build_webmachine_request is the same for each class.

eg.

```ruby

      def build_webmachine_request(rack_req, headers)
        Webmachine::Request.new(rack_req.request_method,
                                rack_req.url,
                                headers,
                                RequestBody.new(rack_req),
                                routing_tokens(rack_req))
      end
```